sql3/planner/inbuiltfunctionsstring.go

```diff
@@ -644,3 +660,34 @@ func (n *callPlanExpression) EvaluateReplicate(currentRow []interface{}) (interf
 	return strings.Repeat(stringArg, int(intArg)), nil
 
 }
+
+// EvaluateFormat function formats according to a format specifier and returns resulting string.
+func (n *callPlanExpression) EvaluateFormat(currentRow []interface{}) (interface{}, error) {
+	// first arg must be a string
+	argEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+	if argEval == nil {
+		return nil, nil
+	}
+	formatString, ok := argEval.(string)
+	if !ok {
+		return nil, sql3.NewErrUnexpectedTypeConversion(0, 0, argEval)
+	}
+
+	var args []interface{}
+
+	// loop, since args can be of any length.
+	for _, arg := range n.args[1:] {
+		argEval, err = arg.Evaluate(currentRow)
+		if err != nil {
+			return nil, err
+		}
+		if argEval == nil {
+			return nil, nil
+		}
```

Comment:
I would argue that this `nil` check should be removed completely.
The `FormatNullArg` test related to this does not do what I would expect; in other words I would expect a `nil` argument to cause the entire string to be returned as `nil`.
If we following `fmt.Sprintf()` logic, that test should return `"this = %!s(<nil>)"`.

Reply:
Alternatively, if we don't want to support `nil` arguments, then that nil check could return an error.

Reply:
The idea here is that, like all other functions, If any arg is null, we return NULL. But, I see that this might not be the case for Format() function. Will change to return error then.